### PR TITLE
Adding a new hook that checks if S3 buckets have versioning enabled

### PR DESCRIPTION
**Pull request by shankar-cloud-builder**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# AWS CloudFormation Community Registry Extensions
+# Static Analysis for Infrastructure as Code to validate against AWS Best Practices
```
Comment (ericzbeard) [inline README.md:1]: This is the top level readme.

File: hooks/S3_BucketVersioningCheck/src/awscommunity_s3_versioningcheck/handlers.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        session: Optional[SessionProxy],
+        request: HookHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+        type_configuration: TypeConfigurationModel
+) -> ProgressEvent:
+    target_model = request.hookContext.targetModel
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS
+    )
+    # TODO: put code here
```
Comment (ericzbeard) [inline hooks/S3_BucketVersioningCheck/src/awscommunity_s3_versioningcheck/handlers.py:64]: Remove TODOs

File: hooks/S3_BucketVersioningCheck/src/awscommunity_s3_versioningcheck/handlers.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        session: Optional[SessionProxy],
+        request: BaseHookHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+        type_configuration: TypeConfigurationModel
+) -> ProgressEvent:
+    target_model = request.hookContext.targetModel
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS
+    )
+    # TODO: put code here
```
Comment (ericzbeard) [inline hooks/S3_BucketVersioningCheck/src/awscommunity_s3_versioningcheck/handlers.py:101]: Remove TODO

File: hooks/S3_BucketVersioningCheck/src/awscommunity_s3_versioningcheck/handlers.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    try:
+        # A Hook that checks to see that S3 bucket's versioning is enabled
+        target_name = request.hookContext.targetName
+        if "AWS::S3::Bucket" == target_name:
+            return _validate_s3_bucket_versioning(request.hookContext.targetModel.get("resourceProperties"))
+        else:
+            raise exceptions.InvalidRequest(f"Unknown target type: {target_name}")
+
+        # Reading the Resource Hook's target current properties and previous properties
+        resource_properties = target_model.get("resourceProperties")
```
Comment (ericzbeard) [inline hooks/S3_BucketVersioningCheck/src/awscommunity_s3_versioningcheck/handlers.py:113]: Will this code ever be reached?

**Review by ericzbeard — CHANGES_REQUESTED** [review]

**Comment by kddejong** [comment]
Closed in favor of the already existing hook of the same purpose.

**Pull request closed without merging**